pkg/stores/proxy/proxy_store.go

```diff
@@ -61,11 +61,11 @@ type ClientGetter interface {
 	TableAdminClientForWatch(ctx *types.APIRequest, schema *types.APISchema, namespace string, warningHandler rest.WarningHandler) (dynamic.ResourceInterface, error)
 }
 
-type warningBuffer struct {
+type WarningBuffer struct {
 	Warnings []types.Warning
 }
```

Comment:
nit: because I don't know the timeline on this, but if you don't make this a struct it makes it easier for callers to use since we would remove the need for a pointer receiver to use a Warning Buffer which is better inline with ['Make the zero value useful.' ](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=385s&ab_channel=TheGoProgrammingLanguage)go proverb
```suggestion
type WarningBuffers []types.Warning
```

Reply:
not sure I follow this just yet.  What does HandleWarningHeader look like for this scenario?